OpenRA.Mods.Common/Traits/Render/DrawLineToTarget.cs

```diff
@@ -63,17 +63,30 @@ void INotifySelected.Selected(Actor self)
 			ShowTargetLines(self);
 		}
 
-		IEnumerable<IRenderable> IRenderAboveShroud.RenderAboveShroud(Actor self, WorldRenderer wr)
+		bool ShouldRender(Actor self)
 		{
 			if (!self.Owner.IsAlliedWith(self.World.LocalPlayer) || Game.Settings.Game.TargetLines == TargetLinesType.Disabled)
-				yield break;
+				return false;
 
 			// Players want to see the lines when in waypoint mode.
 			var force = Game.GetModifierKeys().HasModifier(Modifiers.Shift) || self.World.OrderGenerator is ForceModifiersOrderGenerator;
 
 			if (Game.RunTime > lifetime && !force)
```

Comment:
This could simplify to `return force || Game.RunTime <= lifetime`

Reply:
:+1: